64.md

```diff
@@ -0,0 +1,38 @@
+NIP-64
+======
+
+Trust Rank
+----------
+
+`draft` `optional`
+
+A trust rank is a public way to individually assign a quantifiable level of trust on specifc topics to a nostr identity.
+Ranks can be used to discover trusted people or content on a common topic.
+
+## Event Structure
+
+The `kind:30382` is used to assign public relationship statuses between the event author and a pubkey set to the `d` tag.
+This NIP assigns one or more trust ranks by adding `T` tags with different topic names to such event
+using the following structure: `["T", "<int_value_ranging_from_-3_to_3>:<topic-name>", "<context_dependent_marker>"]`.
+
+Example:
+
+```js
+{
+  "kind": 30382,
+  "tags": [
+    ["d", "<pubkey>"],
+    ["T", "3:wiki:role-playing-video-game", "x"], // has written a based wiki article about RPG
+    ["T", "2:driver"], // polite and skilled driver but doesn't turn the air conditioner on
+    ["T", "3:buyer"], // no problems when selling to this person
+    ["T", "-2:seller"] // had a bad experience when buying from this person or company
```

Comment:
I would prefer a different `kind` for each trust rank application or another indexable tag that sets the application like `["D", "wiki"]`. Otherwise a wiki client would just discard the `driver`, `buyer`, and `seller` tags when updating a wiki value, because these are not useful for the client, leading to interoperability issues.

Reply:
My point remains. Leaving junk data untouched would require client devs to write extra code for no benefit to their own clients. It's also more bandwidth efficient if I can query only the tags that are of interest to me.

Reply:
> Leaving junk data untouched would require client devs to write extra code for no benefit

Even if there was a kind just for ranking authors of wiki articles, the client would have to fetch the latest version and don't touch ratings on topics other than the current one. Similar work.

> [...] indexable tag that sets the application like ["D", "wiki"] [...] It's also more bandwidth efficient if I can query only the tags that are of interest to me

One isn't supposed to fetch all wiki-related trust scores but do one article at a time like this:
```
{ kinds: [30382], authors: ['<my>', '<follows'>, '<...>'], tags: [
  '3:wiki:nostr',
  '2:wiki:nostr',
  '1:wiki:nostr',
]}
```
The above will return just people with positively ranked knowledge on the encyclopedic nostr topic (randed by one's follows). Those with the `d` marker will lead to a recommended article variant. Pubkeys ranked with `r` markers can be used on a follow-up query in the authors filter till reaching a `d` marker to lead to a recommended article variant.

Reply:
> Even if there was a kind just for ranking authors of wiki articles, the client would have to fetch the latest version and don't touch ratings on topics other than the current one. Similar work.

No. All wiki ratings are relevant to a wiki client, every other ranking is junk. Same with the legacy follow lists. If a client does not implement following relays and hashtags than it would just ignore these tags, why would the client cache these irrelevant tags? This leads to clients nuking each others lists. 
Nip51(Lists) does it right, having a different kind for each use-case because managing one giant list is hell, same applies to these rankings.
Also it's much more efficient if you don't have to constantly refetch and validate the event when a marketplace client changes tags that have nothing to do with your wiki use-case.
Doesn't have to be a new kind, maybe put it in the d tag like  `["d", "wiki:<pubkey>"]`